services/streammanager/bqstream/bqstreammanager.go

```diff
@@ -66,24 +66,19 @@ func NewProducer(destinationConfig interface{}, o Opts) (*Client, error) {
 	if err != nil {
 		return nil, createErr(err, "error in BQStream while unmarshalling destination config")
 	}
-	var confCreds []byte
-	if config.Credentials == "" {
-		return nil, createErr(err, "credentials not being sent")
-	}
-	if err = googleutils.CompatibleGoogleCredentialsJSON([]byte(config.Credentials)); err != nil {
-		return nil, createErr(err, "incompatible credentials")
-	}
-	confCreds = []byte(config.Credentials)
-	err = json.Unmarshal(confCreds, &credentialsFile)
-	if err != nil {
-		return nil, createErr(err, "error in BQStream while unmarshalling credentials json")
-	}
 	opts := []option.ClientOption{
-		option.WithCredentialsJSON([]byte(config.Credentials)),
 		option.WithScopes([]string{
 			gbq.BigqueryInsertdataScope,
 		}...),
 	}
+	if !googleutils.IsEmptyCredentials(config.Credentials) {
```

Comment:
We will most probably need an additional `Config` option, e.g. `WorkloadIdentity bool`, so that we don't allow users to skip `Credentials`, unless they have explicitly opted for using GKE Workload Identity as an authentication method.

Reply:
The thing is that, the workload identity changes are being made for a customer. They desire to deploy our data-plane in Google cloud. Since they are waiting for this feature for a long-time, we are testing this feature in AWS. Hence we don't want to add any deployment related stuff to control plane dashboard as of now.

@krishna2020 @saikumarrs @koladilip correct me if I'm wrong here

Reply:
TL;DR it needs to be evident for future generations of developers why we are allowing empty credentials here

Maybe another runtime flag should be passed, eg. through an environment variable for controlling whether we shall allow users to skip credentials or not, e.g. GKE_DEPLOYMENT=true.
I am saying this because if we leave it this way,  a future refactoring might not take this case (GKE) into consideration and change the system's behaviour to always require `Credentials`, breaking workload identity support.

Reply:
@sanpj2292 I agree with @atzoum second suggestion, we should keep control plane agnostic to data plane deployment as it can change right. If we can detect or supply using environment variable about data plane and make decisions about code paths based on it would be good :)

Reply:
Made the changes. Can you guys check and confirm if they're ok ? @aris1009 @koladilip